File: src/ReverseInteger.scala
package leetcode

object ReverseInteger2 extends App {
  def reverse(x: Int): Int = {
     var rev = 0
     var b = x
     while(b != 0){
       val a = b & 0x1
       rev = rev << 1
       rev += a
       b = b >> 1
     }
     if(rev > 0 && x > 0 || rev < 0 && x < 0) rev else 0
  }

  println(reverse(123))
}

File: src/main/scala/ReverseInteger.java
package leetcode;

/**
 * Created by jstalker on 18.06.17.
 */
public class ReverseInteger {
    public static int reverse(int x) {
      long res = 0;
      int a = Math.abs(x);
      while(a > 0){
        int digit = a % 10;
        //shift res to the right
        res = res* 10 + digit;
        if(res > Integer.MAX_VALUE) return 0;
        a = a / 10;
      }
      if(x < 0) return (int)-res;
      return (int)res;
    }

    public static void main(String... args) {
         System.out.println(reverse(1534236469)) ;
    }

}
